### ISSUE TITLE
Training makes model incoherent

### ISSUE THREAD

**Issue by l3utterfly** [axolotl-ai-cloud/axolotl#905]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Train successfully

### Current behaviour

Training leaves the model incoherent.

### Steps to reproduce

1. Use config specified below
2. Set learning rate = 0 to so dataset does not matter
3. train until the 300th checkpoint
4. run inference on the checkpoint model

### Config yaml

base_model: /mnt/m2_4tb/text-generation-webui/models/MiniMA-3B
model_type: LlamaForCausalLM
tokenizer_type: LlamaTokenizer
is_llama_derived_model: true

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: /mnt/m2_4tb/Layla-datasets/datasets_formatted/layla/openchat.topicalchat.sharegpt.jsonl
    ds_type: json # see other options below
    type: sharegpt

dataset_prepared_path: last_run_prepared
val_set_size: 0.05
output_dir: ./out

sequence_len: 2048
sample_packing: true
pad_to_sequence_len: true

adapter:
lora_model_dir:
lora_r:
lora_alpha:
lora_dropout:
lora_target_linear:
lora_fan_in_fan_out:

wandb_project:
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model:

gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 3
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true
flash_attn_cross_entropy: false
flash_attn_rms_norm: true
flash_attn_fuse_qkv: false
flash_attn_fuse_mlp: true

warmup_steps: 100
eval_steps: 0.05
eval_table_size:
save_steps: 300
debug:
deepspeed: deepspeed/zero3.json # multi-gpu only
weight_decay: 0.1
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
I wonder if it's due to not calling post_training on the model, which has just been fixed. Could you retry this on the latest main?

**Comment by NanoCode012** [comment]
I believe I read another issue where you mentioned this #744 . Closing this in favor of that.
